src/main/java/com/tstepnik/planner/controller/TaskController.java

```diff
@@ -0,0 +1,58 @@
+package com.tstepnik.planner.controller;
+
+import com.tstepnik.planner.domain.Task;
+import com.tstepnik.planner.service.TaskService;
+import org.springframework.http.HttpStatus;
+import org.springframework.http.ResponseEntity;
+import org.springframework.security.access.prepost.PreAuthorize;
+import org.springframework.web.bind.annotation.*;
+
+import javax.validation.Valid;
+import java.security.Principal;
+import java.util.List;
+
+@RestController
+@RequestMapping("/api/tasks")
+public class TaskController {
+
+    private final TaskService taskService;
+
+    public TaskController(TaskService taskService) {
+        this.taskService = taskService;
+    }
+
+    @PreAuthorize("hasRole('ADMIN')")
+    @GetMapping("/all")
+    public ResponseEntity<List<Task>> tasks() {
+        return ResponseEntity.ok(taskService.getTasks());
+    }
+
+    @PreAuthorize("hasRole('USER')")
+    @GetMapping
+    public ResponseEntity<List<Task>> userTasks(Principal principal) {
+        List<Task> tasks = taskService.userTasks(principal);
+        return ResponseEntity.ok(tasks);
+    }
+
+    @PostMapping
+    @PreAuthorize("hasRole('USER')")
+    public ResponseEntity<Task> addTask(@Valid @RequestBody Task userTask, Principal principal) {
+        Task task = taskService.addTask(userTask, principal);
+        return new ResponseEntity<>(task, HttpStatus.CREATED);
+    }
+
+    @PutMapping
```

Comment:
Missing id in url

Reply:
I did it especially.

Reply:
?

Reply:
for Put and Post and get is the same global url /tasks.

Reply:
And I am saying that this is wrong. URL for updates and deleting should include id of entity that user wants to delete.